### PR TITLE
feat: update wallet dialog

### PR DESCRIPTION
**Pull request by hanszhoou**
## Description
https://mask.atlassian.net/browse/MF-409

Closes # (NO_ISSUE)

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/mask/src/components/shared/WalletStatusBox/index.tsx [hunk 1]
```diff
@@ -3,107 +3,127 @@ import {
...
+        showWalletName: [ProviderType.MaskWallet, 'Phantom', 'Blocto'].includes(providerDescriptor?.type ?? ''),
+    })
+    const connection = useWeb3Connection(NetworkPluginID.PLUGIN_EVM)
     const chainId = useChainId()
     const account = useAccount()
     const wallet = useWallet()
-    const providerType = useProviderType()
-    const providerDescriptor = useProviderDescriptor()
+    const { value: balance = '0' } = useBalance()
+    const { value: nativeToken } = useNativeToken(NetworkPluginID.PLUGIN_EVM)
```
Comment (guanbinrui) [inline packages/mask/src/components/shared/WalletStatusBox/index.tsx:126]: It cannot suppose the EVM chain is used.

File: packages/mask/src/plugins/Wallet/SNSAdaptor/ConnectWalletDialog/ConnectionProgress.tsx [hunk 2]
```diff
@@ -96,30 +115,32 @@ export function ConnectionProgress(props: ConnectionProgressProps) {
...
-                            ),
-                        }}
-                        values={{
-                            providerName: Others.providerResolver.providerName(providerType),
-                            providerShortenLink: Others.providerResolver.providerShortenLink(providerType),
-                        }}
-                    />
-                </Typography>
-            </Card>
+            {providerDescriptor?.ID === `${NetworkPluginID.PLUGIN_EVM}_walletconnect` ? null : (
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Wallet/SNSAdaptor/ConnectWalletDialog/ConnectionProgress.tsx]:
```suggestion
            {providerDescriptor?.providerType === ProviderType.WalletConnect ? null : (
```

File: packages/mask/src/plugins/Wallet/SNSAdaptor/SelectProviderDialog/PluginProviderRender.tsx [hunk 3]
```diff
@@ -151,20 +175,29 @@ export function PluginProviderRender({
...
+                                        {undeterminedNetworkID === network.ID && (
+                                            <SelectedIcon className={classes.checkedBadge} />
+                                        )}
+                                    </div>
+                                    <Typography
+                                        className={cx(
+                                            classes.networkName,
+                                            undeterminedNetworkID === network.ID ? classes.selected : '',
+                                        )}>
+                                        {network.networkSupporterPluginID === NetworkPluginID.PLUGIN_EVM
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Wallet/SNSAdaptor/SelectProviderDialog/PluginProviderRender.tsx]: All chains should implement a chain resolver.

File: packages/mask/src/components/shared/WalletStatusBox/index.tsx [hunk 4]
```diff
@@ -140,57 +164,64 @@ export function WalletStatusBox(props: WalletStatusBox) {
...
                             className={classes.link}
                             href={Others?.explorerResolver.addressLink?.(chainId, account) ?? ''}
                             target="_blank"
                             title={t('plugin_wallet_view_on_explorer')}
                             rel="noopener noreferrer">
-                            <ExternalLink className={classes.linkIcon} size={14} />
+                            <LinkOutDarkIcon className={classes.icon} />
                         </Link>
                     </div>
+                    {networkDescriptor?.networkSupporterPluginID === NetworkPluginID.PLUGIN_EVM ? (
```
Comment (guanbinrui) [inline packages/mask/src/components/shared/WalletStatusBox/index.tsx:201]: Why only EVM?

File: packages/mask/src/components/shared/WalletStatusBox/index.tsx [hunk 5]
```diff
@@ -140,57 +164,64 @@ export function WalletStatusBox(props: WalletStatusBox) {
...
+                            <Typography className={classes.balance}>
+                                {formatBalance(balance, nativeToken?.decimals, 3)} {nativeToken?.symbol}
+                            </Typography>
+                        </div>
+                    ) : null}
                 </div>
 
                 {!props.disableChange && (
                     <section>
+                        {networkDescriptor?.networkSupporterPluginID === NetworkPluginID.PLUGIN_EVM ? (
```
Comment (guanbinrui) [inline packages/mask/src/components/shared/WalletStatusBox/index.tsx:212]: Why only EVM?

File: packages/mask/src/plugins/Wallet/SNSAdaptor/SelectProviderDialog/PluginProviderRender.tsx [hunk 6]
```diff
@@ -218,3 +230,38 @@ export function PluginProviderRender({
...
+                        <ImageIcon size={30} icon={network.icon} />
+                    </NetworkIconClickBait>
+                ) : (
+                    <ImageIcon size={30} icon={network.icon} />
+                )}
+                {undeterminedNetworkID === network.ID && <SelectedIcon className={classes.checkedBadge} />}
+            </div>
+            <Typography
+                className={cx(classes.networkName, undeterminedNetworkID === network.ID ? classes.selected : '')}>
+                {Others?.chainResolver.chainName(network.chainId as number)}
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Wallet/SNSAdaptor/SelectProviderDialog/PluginProviderRender.tsx]:
```suggestion
                {Others?.chainResolver.chainName(network.chainId)}
```

File: packages/mask/src/plugins/Wallet/SNSAdaptor/SelectProviderDialog/PluginProviderRender.tsx [hunk 7]
```diff
@@ -218,3 +230,38 @@ export function PluginProviderRender({
...
 }
+
+interface NetworkItemProps {
+    onNetworkIconClicked: PluginProviderRenderProps['onNetworkIconClicked']
+    NetworkIconClickBait?: PluginProviderRenderProps['NetworkIconClickBait']
+    network: Web3Helper.NetworkDescriptorAll
+    undeterminedNetworkID: string | undefined
+}
+
+function NetworkItem({ onNetworkIconClicked, NetworkIconClickBait, network, undeterminedNetworkID }: NetworkItemProps) {
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Wallet/SNSAdaptor/SelectProviderDialog/PluginProviderRender.tsx:239]: So where is the `<ShadowRootTooltip />`?

Comment (hanszhoou) [reply packages/mask/src/plugins/Wallet/SNSAdaptor/SelectProviderDialog/PluginProviderRender.tsx:239]: Tooltip is removed by the latest design.

File: packages/mask/src/components/shared/WalletStatusBox/index.tsx (hunk 1 above)
Comment (hanszhoou) [reply packages/mask/src/components/shared/WalletStatusBox/index.tsx:126]: not implemented yet.

File: packages/mask/src/components/shared/WalletStatusBox/index.tsx (hunk 4 above)
Comment (hanszhoou) [reply packages/mask/src/components/shared/WalletStatusBox/index.tsx:201]: not implemented yet.

File: packages/mask/src/plugins/Wallet/SNSAdaptor/SelectProviderDialog/PluginProviderRender.tsx (hunk 7 above)
Comment (guanbinrui) [reply packages/mask/src/plugins/Wallet/SNSAdaptor/SelectProviderDialog/PluginProviderRender.tsx:239]: Ack

**Pull request merged**